File: Qb.Net/Column.cs
using Viten.QueryBuilder.SqlOm;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Viten.QueryBuilder
{
  /// <summary>
  /// Класс колонки запроса
  /// </summary>
  public class Column
  {
    internal SelectColumn Col;

    /// <summary>Конструктор</summary>
    internal Column(string columnName)
    {
      Col = new SelectColumn(columnName);
    }

    /// <summary>Конструктор</summary>
    internal Column(string columnName, From table)
    {
      Col = new SelectColumn(columnName, table != null ? table.Term : null);
    }

    /// <summary>Конструктор</summary>
    internal Column(string columnName, From table, string columnAlias)
    {
      Col = new SelectColumn(columnName, table != null ? table.Term : null, columnAlias);
    }

    /// <summary>Конструктор</summary>
    internal Column(string columnName, string columnAlias, AggFunc function)
    {
      Col = new SelectColumn(columnName, null, columnAlias, function);
    }

    /// <summary>Конструктор</summary>
    internal Column(string columnName, string columnAlias, From table, AggFunc function)
    {
      Col = new SelectColumn(columnName, table != null ? table.Term : null, columnAlias, function);
    }

    internal Column(Expr expr, string columnAlias)
    {
      Col = new SelectColumn(expr.Expression, columnAlias);
    }

    /// <summary>Определить колонку</summary>
    public static Column New(string columnName, string columnAlias, AggFunc function)
    {
      return new Column(columnName, columnAlias, function);
    }

    /// <summary>Определить колонку</summary>
    public static Column New(string columnName)
    {
      return new Column(columnName);
    }

    /// <summary>Определить колонку</summary>
    public static Column New(string columnName, From table)
    {
      return new Column(columnName, table);
    }

    /// <summary>Определить колонку</summary>
    public static Column New(string columnName, From table, string columnAlias)
    {
      return new Column(columnName, table, columnAlias);
    }

    /// <summary>Определить колонку</summary>
    public static Column New(string columnName, string columnAlias, From table, AggFunc function)
    {
      return new Column(columnName, columnAlias, table, function);
    }

    /// <summary>Определить колонку</summary>
    public static Column New(string columnName, AggFunc function)
    {
      return new Column(columnName, null, null, function);
    }

    /// <summary>Определить колонку</summary>
    public static Column New(string columnName, From table, AggFunc function)
    {
      return new Column(columnName, null, table, function);
    }

    /// <summary>Определить колонку</summary>
    public static Column New(Expr expr, string columnAlias)
    {
      return new Column(expr, columnAlias);
    }

  }
}

File: Qb.Js/src/Column.ts
import { SelectColumn } from "./SelectColumn";
import { AggFunc } from "./Enums";
import { From } from "./From";
import { FromTerm } from "./FromTerm";
import { Utils } from "./Utils";
import { Expr } from "./Expr";

export class Column {
  private Col: SelectColumn;

  static New(columnName: string): Column;
  static New(columnName: string | Expr, columnAlias: string): Column;
  static New(columnName: string, columnAlias: string, func: AggFunc): Column;
  static New(columnName: string, fromTable: From): Column;
  static New(columnName: string, columnAlias: string, fromTable: From): Column;
  static New(columnName: string, columnAlias: string, fromTable: From, func: AggFunc): Column;
  static New(): Column {
    var columnName, columnAlias: string | undefined;
    var fromTable: From | undefined;
    var func: AggFunc | undefined;
    var expt: Expr | undefined;
    if (arguments.length >= 1) {
      columnName = arguments[0];
    }
    if (arguments.length >= 2) {
      if (typeof arguments[1] == "string") {
        columnAlias = arguments[1];
      } else {
        fromTable = arguments[1];
      }
    }
    if (arguments.length >= 3) {
      if (typeof arguments[2] == "number") {
        func = arguments[2];
      } else {
        fromTable = arguments[2];
      }
    }
    if (arguments.length >= 4) {
      func = arguments[3];
    }
    var fromTerm: FromTerm | undefined = fromTable instanceof From ? <FromTerm>fromTable.Term : undefined;
    var retVal = new Column();
    retVal.Col = new SelectColumn(columnName, columnAlias, fromTerm, func);
    return retVal;
  }
}